examples/ota-provider-app/linux/main.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
+
+#include <app/clusters/ota-provider/ota-provider-delegate.h>
+#include <app/clusters/ota-provider/ota-provider.h>
+#include <app/server/Server.h>
+#include <app/util/util.h>
+#include <core/CHIPError.h>
+#include <support/CHIPArgParser.hpp>
+#include <support/CHIPMem.h>
+#include <support/RandUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+#include "BdxOtaSender.h"
+#include "OTAProviderExample.h"
+
+#include <fstream>
+#include <iostream>
+#include <unistd.h>
+
+using namespace chip;
+using namespace chip::ArgParser;
+
+using chip::app::clusters::OTAProviderDelegate;
+
+constexpr uint16_t kOptionFilepath = 'f';
+
+const char * gOtaFilepath = nullptr;
+
+// Arbitrary BDX Transfer Params
+constexpr uint32_t kMaxBdxBlockSize = 1024;
+constexpr uint32_t kBdxTimeoutMs    = 5 * 60 * 1000; // OTA Spec mandates >= 5 minutes
+constexpr uint32_t kBdxPollFreqMs   = 500;
+
+bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier, const char * aName, const char * aValue)
+{
+    bool retval = true;
+
+    switch (aIdentifier)
+    {
+    case kOptionFilepath:
+        if (0 == access(aValue, R_OK))
```

Comment:
I thought access() returns 0 on success.  Am I missing something here?

Reply:
Thanks for catching this - you're correct.